insalan/payment/models.py

```diff
@@ -0,0 +1,278 @@
+        max_length=20,
+        blank=False,
+        null=False,
+        verbose_name=_("Catégorie de produit"),
+        default=ProductCategory.PIZZA,
+        choices=ProductCategory.choices,
+    )
+    associated_tournament = models.ForeignKey(
+        Tournament,
+        on_delete=models.CASCADE,
+        verbose_name=_("Tournoi associé"),
+        null=True,
+        blank=True,
+    )
+    available_from = models.DateTimeField(
+        blank=True,
+        null=False,
+        default=timezone.now,
+        verbose_name=_("Disponible à partir de"),
+    )
+    available_until = models.DateTimeField(
+        null=False, verbose_name=_("Disponible jusqu'à")
+    )
+
+    def can_be_bought_now(self) -> bool:
+        """Returns whether or not the product can be bought now"""
+        return self.available_from <= timezone.now() <= self.available_until
+
+
+class Transaction(models.Model):
+    """
+    A transaction is a record from helloasso intent.
+
+    A transaction cannot exist alone and should be used only to reflect helloasso payment
+    """
+
+    class Meta:
+        """Meta information"""
+
+        ordering = ["-last_modification_date"]
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+    payer = models.ForeignKey(
+        User, null=True, on_delete=models.SET_NULL, verbose_name=_("Utilisateur")
+    )
+    products = models.ManyToManyField(
+        Product, through="ProductCount"
+    )  # A transaction can be composed of n products
+    payment_status = models.CharField(
+        max_length=10,
+        blank=True,
+        default=TransactionStatus.PENDING,
+        choices=TransactionStatus.choices,
+        null=False,
+        verbose_name=_("État de la Transaction"),
+    )
+    creation_date = models.DateTimeField(verbose_name=_("Date de creation"))
+    last_modification_date = models.DateTimeField(
+        verbose_name=_("Date de dernière modification")
+    )
+    intent_id = models.IntegerField(
+        blank=False,
+        null=True,
+        editable=False,
+        verbose_name=_("Identifiant de paiement"),
+    )
+    amount = models.DecimalField(
+        null=False,
+        default=0.00,
+        max_digits=5,
+        decimal_places=2,
+        verbose_name=_("Montant"),
+    )
+
+    @staticmethod
+    def new(**data):
+        """create a new transaction based on products id list and a payer"""
+        logger.debug(f"in the constructor {data}")
+        fields = {}
+        fields["creation_date"] = timezone.make_aware(datetime.now())
+        fields["last_modification_date"] = fields["creation_date"]
+        fields["payer"] = data["payer"]
+        transaction = Transaction.objects.create(**fields)
+        data["products"].sort(key=lambda x: int(x.id))
+        for pid, grouper in itertools.groupby(data["products"]):
+            # Validate that the products can be bought
+            if not pid.can_be_bought_now():
+                raise ValidationError(
```

Comment:
same remark about f-string here

Reply:
Done for all usage in the files of payment